packages/synapse-react-client/src/mocks/msw/handlers/asyncJobHandlers.ts

```diff
@@ -0,0 +1,126 @@
+import { DefaultBodyType, rest } from 'msw'
+import {
+  BackendDestinationEnum,
+  getEndpoint,
+} from '../../../utils/functions/getEndpoint'
+import { uniqueId } from 'lodash-es'
+import {
+  AsynchronousJobStatus,
+  AsyncJobId,
+} from '@sage-bionetworks/synapse-types'
+import { ASYNCHRONOUS_JOB_TOKEN } from '../../../utils/APIConstants'
+
+/**
+ * Global mapping between async job token and request/response. This sort of acts as an in-memory database for asynchronous jobs.
+ */
+const mapOfAsyncJobs = new Map<
+  string,
+  {
+    request: unknown
+    response: unknown
+  }
+>()
+
+/**
+ * Generates MSW handlers for asynchronous jobs.
+ * @param requestPath the path to the service-specific asynchronous job request endpoint.
+ *   For example, `'/repo/v1/entity/:id/table/query/async/start'`
+ * @param responsePath a function that takes the token parameter ID and returns a path to the service-specific asynchronous job response endpoint.
+ *   For example, `param => /repo/v1/entity/:id/table/query/async/get/${param}`
+ *   The generic asynchronous job retrieval handler will be generated automatically.
+ * @param responseBody the response body, or a function that takes the request body and returns the response body
+ * @param backendOrigin the backend origin to use for the handlers.
+ */
+export function generateAsyncJobHandlers<
```

Comment:
Before this change, we weren't able to mock out multiple types of asynchronous jobs, like table query + fetching annotation columns for a given scope. Now we can.